### PR TITLE
double-write selector_id to ticks, jobs, instigators 3/5

### PR DESCRIPTION
**Pull request by prha**
## Summary
This PR adds a required data migration (will be run at the same time as the schema migration).

This ensure that selector_id should be populated for the legacy `jobs` table.  This should help
maintain the correctness of the new `instigators` table (we have to keep track of deletes).

## Test Plan
new BK tests for sqlite, postgres, mysql

### CODE REVIEW

**Comment by prha** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7182** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7182" target="_blank">Graphite</a> 
    * **PR #7185** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7185" target="_blank">Graphite</a> 
      * **PR #7191** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7191" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: python_modules/dagster/dagster/core/host_representation/selector.py [hunk 1]
```diff
@@ -136,13 +137,14 @@ def from_graphql_input(graphql_data):
         )
 
 
-class InstigationSelector(
+@whitelist_for_serdes
+class InstigatorSelector(
```
Comment (gibsondan) [inline python_modules/dagster/dagster/core/host_representation/selector.py:141]: hooray!

File: python_modules/dagster/dagster/core/host_representation/external.py [hunk 2]
```diff
@@ -490,6 +496,12 @@ def get_external_origin(self):
     def get_external_origin_id(self):
         return self.get_external_origin().get_id()
 
+    @property
+    def selector_id(self):
+        if self._selector_id is None:
+            self._selector_id = create_snapshot_id(self._selector)
+        return self._selector_id
```
Comment (gibsondan) [inline python_modules/dagster/dagster/core/host_representation/external.py:503]: is this expensive enough that this cache is useful?

File: python_modules/dagster/dagster/core/storage/event_log/sql_event_log.py [hunk 3]
```diff
@@ -433,14 +433,14 @@ def _apply_migration(self, migration_name, migration_fn, print_fn, force):
         if self.has_secondary_index(migration_name):
             if not force:
                 if print_fn:
-                    print_fn("Skipping already reindexed summary: {}".format(migration_name))
+                    print_fn("Skipping already applied migration: {}".format(migration_name))
```
Comment (gibsondan) [inline python_modules/dagster/dagster/core/storage/event_log/sql_event_log.py:436]: data migration

File: python_modules/dagster/dagster/core/storage/runs/sql_run_storage.py [hunk 4]
```diff
@@ -823,6 +823,8 @@ def _execute_data_migrations(
         for migration_name, migration_fn in migrations.items():
             if self.has_built_index(migration_name):
                 if not force_rebuild_all:
+                    if print_fn:
+                        print_fn("Skipping already applied migration: {}".format(migration_name))
```
Comment (gibsondan) [inline python_modules/dagster/dagster/core/storage/runs/sql_run_storage.py:827]: data migration

File: python_modules/dagster/dagster/core/storage/schedules/migration.py [hunk 5]
```diff
@@ -1,4 +1,57 @@
...
+    it to the jobs table.
+    """
+
+    if print_fn:
+        print_fn("Querying storage.")
+
+    instigator_states = storage.all_instigator_state()
+    for state in tqdm(instigator_states):
+        # should write the selector_id to the jobs table
+        storage.update_instigator_state(state)
```
Comment (gibsondan) [inline python_modules/dagster/dagster/core/storage/schedules/migration.py:30]: does this update any timestamps? That's probably fine as long as it doesn't change the start timestamp or anything

File: python_modules/libraries/dagster-mysql/dagster_mysql_tests/compat_tests/test_back_compat.py [hunk 6]
```diff
@@ -70,3 +70,64 @@ def test_instigators_table_backcompat(hostname, conn_string):
...
+        file_relative_path(__file__, "snapshot_0_14_6_post_schema_pre_data_migration.sql"),
+    )
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        with open(file_relative_path(__file__, "dagster.yaml"), "r") as template_fd:
+            with open(os.path.join(tempdir, "dagster.yaml"), "w") as target_fd:
+                template = template_fd.read().format(hostname=hostname)
+                target_fd.write(template)
+
+        instance = DagsterInstance.from_config(tempdir)
```
Comment (gibsondan) [inline python_modules/libraries/dagster-mysql/dagster_mysql_tests/compat_tests/test_back_compat.py:93]: with DagsterInstance... as instance:

File: python_modules/libraries/dagster-postgres/dagster_postgres_tests/compat_tests/test_back_compat.py [hunk 7]
```diff
@@ -451,3 +451,66 @@ def test_instigators_table_backcompat(hostname, conn_string):
...
+        ),
+    )
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        with open(file_relative_path(__file__, "dagster.yaml"), "r") as template_fd:
+            with open(os.path.join(tempdir, "dagster.yaml"), "w") as target_fd:
+                template = template_fd.read().format(hostname=hostname)
+                target_fd.write(template)
+
+        instance = DagsterInstance.from_config(tempdir)
```
Comment (gibsondan) [inline python_modules/libraries/dagster-postgres/dagster_postgres_tests/compat_tests/test_back_compat.py:476]: with .. as instance:

**Review by gibsondan — APPROVED** [review]

File: python_modules/dagster/dagster/core/host_representation/external.py (hunk 2 above)
Comment (prha) [reply python_modules/dagster/dagster/core/host_representation/external.py:503]: perhaps not... i guess I just trained myself to be wary of it.  I can remove for simplicity and add back if necessary.

File: python_modules/dagster/dagster/core/storage/schedules/migration.py (hunk 5 above)
Comment (prha) [reply python_modules/dagster/dagster/core/storage/schedules/migration.py:30]: it doesn't but it probably should... good catch, I'll rewrite the migration and maybe in a separate diff will change writing the update_timestamp

**Pull request merged**